tiledb/sm/subarray/subarray.cc

```diff
@@ -275,6 +275,59 @@ Status Subarray::add_range(
       dim_idx, Range(&range[0], 2 * coord_size), err_on_range_oob_);
 }
 
+Status Subarray::add_point_ranges(
+    unsigned dim_idx, const void* start, uint64_t count) {
+  if (dim_idx >= this->array_->array_schema_latest()->dim_num())
+    return LOG_STATUS(
+        Status::SubarrayError("Cannot add range; Invalid dimension index"));
+
+  QueryType array_query_type;
+  RETURN_NOT_OK(array_->get_query_type(&array_query_type));
+  if (array_query_type == tiledb::sm::QueryType::WRITE) {
+    if (!array_->array_schema_latest()->dense()) {
+      return LOG_STATUS(Status::SubarrayError(
+          "Adding a subarray range to a write query is not "
+          "supported in sparse arrays"));
+    }
+    if (this->is_set(dim_idx))
+      return LOG_STATUS(
+          Status::SubarrayError("Cannot add range; Multi-range dense writes "
+                                "are not supported"));
+  }
+
+  if (start == nullptr)
+    return LOG_STATUS(
+        Status::SubarrayError("Cannot add ranges; Invalid start pointer"));
+
+  if (this->array_->array_schema_latest()
+          ->domain()
+          ->dimension(dim_idx)
+          ->var_size())
+    return LOG_STATUS(
+        Status::SubarrayError("Cannot add range; Range must be fixed-sized"));
+
+  // Prepare a temp range
+  std::vector<uint8_t> range;
+  uint8_t coord_size =
+      this->array_->array_schema_latest()->dimension(dim_idx)->coord_size();
+  range.resize(2 * coord_size);
+
+  for (size_t i = 0; i < count; i++) {
```

Comment:
Before this you can add a `ranges_[dim_idx].reserve(ranges_[dim_idx].size() + count)` This might avoid additional allocations for the ranges vector.

Reply:
✔️ updated